### PR TITLE
Added tests for HeroMapper class

### PR DESCRIPTION
**Pull request by Arturnix**
Added tests

### CODE REVIEW

File: src/main/java/com/diablo3CharViewer/data_models/HeroDataModel.java [hunk 1]
```diff
@@ -43,6 +43,38 @@ public HeroDataModel(int id, String name, String classHero, int level, int parag
         this.stats = stats;
     }
 
+    public int getId() {
```
Comment (blazejprzyluski) [inline src/main/java/com/diablo3CharViewer/data_models/HeroDataModel.java:46]: Wild thought: can you make all those fields in the class final? I don't really see a need for them to be mutable ATM

File: src/test/java/HeroMapperTest.java [hunk 2]
```diff
@@ -1,2 +1,184 @@
...
+@ExtendWith(MockitoExtension.class)
 public class HeroMapperTest {
+
+    @Mock
+    private FetchToken testFetchTokenMock;
+    @Mock
+    private HeroMapper testHeroMapperMock;
+    @Mock
+    private Map<String, Integer> heroStatsMock;
+    private final FetchToken tokenAsNull = null;
```
Comment (blazejprzyluski) [inline src/test/java/HeroMapperTest.java:27]: Some of those fields are repeated between tests. Can we make them shareable somehow?

File: src/test/java/HeroMapperTest.java [hunk 3]
```diff
@@ -1,2 +1,184 @@
...
+    private HeroMapper testHeroMapper = new HeroMapper();
+    private final FetchToken testFetchToken = new FetchToken();
+    private final String battleTag = "Jokefish#2265";
+    private final String battleTagAsWrongFormat = "abc123";
+    private final String battleTagAsNull = null;
```
Comment (blazejprzyluski) [inline src/test/java/HeroMapperTest.java:32]: Why null?

File: src/test/java/HeroMapperTest.java [hunk 4]
```diff
@@ -1,2 +1,184 @@
...
+    private final String battleTagAsDosentExist = "dsf#123";
+    private final String wrongBattleTagFormatWarning = "Niepoprawny format battleTag! Spróbuj ponownie.";
+    private final String wrongHeroIdFormatWarning = "Niepoprawny format heroId - tylko cyfry! Spróbuj ponownie.";
+    private final String heroId = "162864678";
+    private final int expectedHeroId = 162864678;
+    private final String heroIdAsWrongFormat = "abc123";
+    private final String heroIdAsNull = null;
```
Comment (blazejprzyluski) [inline src/test/java/HeroMapperTest.java:39]: Why null?

File: src/test/java/HeroMapperTest.java [hunk 5]
```diff
@@ -1,2 +1,184 @@
...
+    public void correctHeroStatsFetchedToDataModel() {
+        Mockito.when(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock)).thenReturn(heroDataModel);
+        Assertions.assertTrue(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock).getStats().containsKey("life"));
+        Assertions.assertTrue(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock).getStats().containsValue(437161));
+        Assertions.assertTrue(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock).getStats().containsKey("damage"));
+        Assertions.assertTrue(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock).getStats().containsValue(276225));
+    }
+
+    @Test
+    public void correctHeroStatsMockFetched() { //czy tak jest sens robic skoro na sztywno podaję takie warunki, ze zawsze i tak wyjdzie true?
```
Comment (blazejprzyluski) [inline src/test/java/HeroMapperTest.java:94]: Nope, no point

File: src/test/java/HeroMapperTest.java [hunk 6]
```diff
@@ -1,2 +1,184 @@
...
+
+        String expectedMessage = "argument \"content\" is null";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void providedTokenNullThrowsException() {
+        Exception exception = Assertions.assertThrows(NullPointerException.class, ()-> {
```
Comment (blazejprzyluski) [inline src/test/java/HeroMapperTest.java:175]: assertThrows suffices for exception tests

File: src/main/java/com/diablo3CharViewer/data_models/HeroDataModel.java (hunk 1 above)
Comment (Arturnix) [reply src/main/java/com/diablo3CharViewer/data_models/HeroDataModel.java:46]: done

File: src/test/java/HeroMapperTest.java (hunk 3 above)
Comment (Arturnix) [reply src/test/java/HeroMapperTest.java:32]: now those null initialized variables are put directly to methods

File: src/test/java/HeroMapperTest.java (hunk 4 above)
Comment (Arturnix) [reply src/test/java/HeroMapperTest.java:39]: done

File: src/test/java/HeroMapperTest.java (hunk 6 above)
Comment (Arturnix) [reply src/test/java/HeroMapperTest.java:175]: done

File: src/test/java/HeroMapperTest.java (hunk 5 above)
Comment (Arturnix) [reply src/test/java/HeroMapperTest.java:94]: deleted

File: src/test/java/HeroMapperTest.java (hunk 2 above)
Comment (Arturnix) [reply src/test/java/HeroMapperTest.java:27]: Make them protected in separate partent class

File: src/test/java/AccountMapperTest.java [hunk 7]
```diff
@@ -14,22 +13,14 @@
 import static org.mockito.Mockito.lenient;
 
 @ExtendWith(MockitoExtension.class)
-public class AccountMapperTest {
+public class AccountMapperTest extends ShareableDataForTests {
```
Comment (blazejprzyluski) [inline src/test/java/AccountMapperTest.java:16]: This is wrong use of inheritance. Child class should be a subclass of the parent, not just use the parameters.

**Pull request merged**